editor/src/clj/editor/system.clj

```diff
@@ -75,6 +75,14 @@
 
 (defn defold-unpack-path
   ^String []
+  ;; This call ensures we have unpacked all the required libraries and binaries
+  ;; so that they are ready to use. It contains a check so that it will only do
+  ;; this the first time the method is called. It is safe to call from any
+  ;; thead, but will block until the unpacking thread has completed.
+  ;;
+  ;; Having this call here mainly benefits the tests and repl-interactions, as
+  ;; the editor will also explicitly call unpackResources at startup.
```

Comment:
Have you considered using `delay` that unpacks resources and is dereferenced here and on editor startup so you don't have to write locks?

Reply:
I don't think I can, because there are Java classes like `ParticleLibrary.java` that call this in their static initializer. In the editor, we make a blocking call to `unpackResources` before we even start to load the Clojure namespaces.

Reply:
Ah, makes sense!